Cookbook generation crashes using empty meta example files

Currently, when an empty meta example file is referenced for a cookbook the generation process crashes.

``Generating LALR tables
Generating LALR tables
Generating LALR tables
Generating LALR tables
Translating clustering/hierarchical.sg
Translating clustering/gmm.sg
Translating gaussian_processes/gaussian_process_regression.sg
Translating distance/tanimoto_distance.sg
Traceback (most recent call last):
  File "/home/vagrant/shogun/examples/meta/generator/generate.py", line 120, in <module>
    includedTargets=args.targets, storeVars=storeVars)
  File "/home/vagrant/shogun/examples/meta/generator/generate.py", line 52, in translateExamples
    ast = parse(file.read(), os.path.join(dirRelative, filename))
  File "/home/vagrant/shogun/examples/meta/generator/parse.py", line 241, in parse
    return parser.parse(programString, filePath)
  File "/home/vagrant/shogun/examples/meta/generator/parse.py", line 28, in parse
    if programString[-1] != "\n":
IndexError: string index out of range
ninja: build stopped: subcommand failed.``

--- a/examples/meta/generator/parse.py
+++ b/examples/meta/generator/parse.py
@@ -25,7 +25,7 @@ class FastParser:
         """
 
         # Add trailing endline if missing
-        if programString[-1] != "\n":
+        if len(programString) and programString[-1] != "\n":
             programString += "\n"
 
         # Parse program and add FilePath key to object

